src/storage/src/hummock/store/version.rs

```diff
@@ -12,44 +12,57 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::collections::BTreeMap;
+use std::cmp::Ordering;
+use std::collections::VecDeque;
 use std::ops::RangeBounds;
 use std::sync::Arc;
 
+use itertools::Itertools;
+use risingwave_hummock_sdk::compaction_group::hummock_version_ext::HummockVersionExt;
+use risingwave_hummock_sdk::key::{key_with_epoch, next_key, user_key};
 use risingwave_hummock_sdk::LocalSstableInfo;
-use risingwave_pb::hummock::{HummockVersion, HummockVersionDelta};
+use risingwave_pb::hummock::{HummockVersion, HummockVersionDelta, LevelType};
 
+// use super::memtable::Memtable;
 use super::{GetFutureTrait, IterFutureTrait, ReadOptions};
+use crate::hummock::local_version::{PinnedVersion, PinnedVersionGuard};
+use crate::hummock::shared_buffer::shared_buffer_batch::SharedBufferBatch;
 use crate::hummock::sstable_store::SstableStoreRef;
-use crate::hummock::{HummockResult, HummockStateStoreIter};
-use crate::table::streaming_table::mem_table::MemTable;
+use crate::hummock::utils::{filter_single_sst, prune_ssts, range_overlap};
+use crate::hummock::{get_from_batch, get_from_table, HummockResult, HummockStateStoreIter};
+use crate::monitor::{StateStoreMetrics, StoreLocalStatistic};
+
+type ImmutableMemtable = SharedBufferBatch;
+type ImmIdVec = Vec<usize>;
```

Comment:
Here we may do `type ImmId = SharedBufferBatchId` and store `ImmIdVec` as `Vec<ImmId>`.